### PR TITLE
Merges Kubernetes features into GCT plugin

### PR DESCRIPTION
**Pull request by etanshaul**
### What this PR does
Merges `google-container-tools-intellij` into `google-cloud-intellij`, **preserving Git history**.

### Status
The Kubernetes features are merged into the GCT plugin. All features currently run side by side:
screen shot 2019-02-17 at 2 53 14 pm

### Remaining TODOs
- [x] merge code with history preserved
- [x] get functionality from both plugins working together in the same plugin
- [x] Get all tests to pass
- [x] Java and Kotlin language support plugin-wide
- [x] Get build to fully pass with static analytics (including configuring spotless for both Java & Kotlin)
- [x] bump supported IDE version to minimum supported by the container-tools plugin (2018.2)
- [x] set IDEA dev target version to 2018.3
- [x] delete old Kokoro project configs
- [x] update CI to build correct versions
- [x] rename the container-tools module to `kubernetes`
_TODOs in next PRs_
- [ ] (subsequent PR) rename 'com.google.container..' package naming to 'com.google.kubernetes..'; Note: do not do this until all outstanding PRs in the other branch have been closed and synced with this repo
- [ ] (subsequent PR) factor out the container tools `core` and `common-test-lib` modules to be shared plugin-wide
- [ ] (subsequent PR) refactor documentation (currently all container tools documentation lives in the sub-module)
- [ ] (subsequent PR) GCS panel hidden by default
- [ ] (subsequent PR) disable/feature-flag container-tools feature 
- [ ] (subsequent PR) refactor the action tools menu 
- [ ] (subsequent PR) refactor telemetry according to container-tools strategy

Testing (manual)
- [x] features plugin-wide
- [x] telemetry, from both plugin code
- [x] error reporting, including from Kotlin sources

**IMPORTANT: Prior to merging to master, GitHub commit squashing needs to be temporarily disabled to preserve history of the container-tools repo**

### CODE REVIEW

**Comment by etanshaul** [comment]
@ivanporty ready for review.

This performs the bulk of the work required to merge the container-tools plugin into GCT.

Apologies for the large PR, but this was one of those rare instances where it would have not been worth it to try and split it up. Regardless most of the changes are just bringing the files over from the other repo here with no changes.

**Comment by etanshaul** [comment]
+patflynn just as fyi

File: gradle.properties [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-ideaEdition = IU
+ideaEdition = IC
```
Comment (ivanporty) [inline gradle.properties]: Does it make more sense to leave it IU for combined functionality? It feels we have too many features for IU from app engine and they could be essentially ignored unless we keep changing this every time. We probably need to configure CI to build with both IC/UI?

**Comment by ivanporty** [comment]
LGTM in general, just one small comment, otherwise pretty straightforward merge of modules. Build time though :) this is what I got for my first run - `BUILD SUCCESSFUL in 7m 55s`. Need to refactor those heavy tests.

BTW, let me take the package name refactoring thing (next PR) if you don't mind - wanted to get a feel for the new structure and make sure I know where all the code went.

**Comment by etanshaul** [comment]
sounds good. I did notice an issue manually testing that I'm trying to resolve (an exception during new project creation wizard).

wrt to the package name refactoring, I was thinking we could wait until all open PRs on container tools are merged into this repo (currently there is 1 outstanding). Keeping the package names the same makes it easier to do this (via a patch, or cherry picking).

**Comment by ivanporty** [comment]
Sure, let's do it after that one is merged.

File: gradle.properties (hunk 1 above)
Comment (etanshaul) [reply gradle.properties]: yeah. makes sense. will revert

**Pull request merged**